addon/serializers/application.js

```diff
@@ -86,9 +86,18 @@ export default DS.RESTSerializer.extend({
         // the links property so the adapter can async call the
         // relationship.
         // The adapter findHasMany has been overridden to make use of this.
-        if ( options.relation ) {
-          hash.links = {};
-          hash.links[key] = { type: relationship.type, key: key };
+        if(options.relation) {
+          // hash[key] contains the response of Parse.com: eg {__type: Relation, className: MyParseClassName}
+          // this is an object that make ember-data fail, as it expects nothing or an array ids that represent the records
+          hash[key] = [];
+
+          // ember-data expects the link to be a string
+          // The adapter findHasMany will parse it
+          if (!hash.links) {
+            hash.links = {};
+          }
+
+          hash.links[key] = JSON.stringify({typeKey: relationship.type.typeKey, key: key});
```

Comment:
`typeKey` is an Ember-Data internal string version of the model name. Creating a payload for push that looks like:

```
{
  typeKey: 'someFoo'
}
```

Would not actually set the `typeKey`. Instead polymorphic payload are intended to pass their type in a payload as `type`:

```
{
  type: 'someFoo'
}
```

In short, pushing `typeKey:` into the store doesn't so anything afaik.